core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -216,19 +223,26 @@ object ConsoleConsumer extends Logging {
     * are conflicting.
     */
   def setAutoOffsetResetValue(config: ConsumerConfig, props: Properties) {
+    val (earliestConfigName, latestConfigName) = if (config.useOldConsumer)
+      (OffsetRequest.SmallestTimeString, OffsetRequest.LargestTimeString)
+    else
+      ("earliest", "latest")
+
     if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
       // auto.offset.reset parameter was specified on the command line
-      if (config.options.has(config.resetBeginningOpt) && "latest".equals(props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG))) {
+      if (config.options.has(config.resetBeginningOpt) && earliestConfigName != props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
```

Comment:
Could it be null? We have the `containsKey` check above. Is there a way to pass null explicitly? Even if there is, I'd say the check is still valid since the user has attempted to override `auto.offset.reset` while also providing `--from-beginning`. 

I will update the error message.